src/daemon/message_hexdump.go

```diff
@@ -0,0 +1,141 @@
+package daemon
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/skycoin/skycoin/src/cipher"
+	"github.com/skycoin/skycoin/src/cipher/encoder"
+	"github.com/skycoin/skycoin/src/daemon/gnet"
+)
+
+var registered = false
+
+func serializeMessage(msg gnet.Message) []byte {
+	t := reflect.ValueOf(msg).Elem().Type()
+	msgID := gnet.MessageIDMap[t]
+	bMsg := encoder.Serialize(msg)
+
+	// message length
+	bLen := encoder.SerializeAtomic(uint32(len(bMsg) + len(msgID)))
+	m := make([]byte, 0)
+	m = append(m, bLen...)     // length prefix
+	m = append(m, msgID[:]...) // message id
+	m = append(m, bMsg...)     // message bytes
+	return m
+}
+
+func getSliceContentsString(sl []string, offset int) string {
+	var res string = ""
+	var counter int = 0
+	var currentOff = offset
+	if offset != -1 {
+		var hex = strconv.FormatInt(int64(offset), 16)
+		var l = len(hex)
+		for i := 0; i < 4-l; i++ {
+			hex = "0" + hex
+		}
+		hex = "0x" + hex
+		res += hex + " | "
+	}
+	for i := 0; i < len(sl); i++ {
+		counter++
+		res += sl[i] + " "
+		if counter == 16 {
+			res += "\n"
+			currentOff += 16
+			if offset != -1 {
+				//res += "         " //9 spaces
+				var hex = strconv.FormatInt(int64(currentOff), 16)
+				var l = len(hex)
+				for i := 0; i < 4-l; i++ {
+					hex = "0" + hex
+				}
+				hex = "0x" + hex
+				res += hex + " | "
+			}
+			counter = 0
+		}
+	}
+	for i := 0; i < (16 - counter); i++ {
+		res += "..."
+	}
+	res += "..."
+	return res
+}
+
+func printLHexDumpWithFormat(offset int, name string, buffer []byte) {
+	var hexBuff = make([]string, len(buffer))
+	for i := 0; i < len(buffer); i++ {
+		hexBuff[i] = strconv.FormatInt(int64(buffer[i]), 16)
+	}
+	for i := 0; i < len(buffer); i++ {
+		if len(hexBuff[i]) == 1 {
+			hexBuff[i] = "0" + hexBuff[i]
+		}
+	}
+	fmt.Println(getSliceContentsString(hexBuff, offset), name)
+}
+
+func GenerateRandomSha256() cipher.SHA256 {
```

Comment:
should not be a public method in the package, and is only used by tests.  Use `testutil.RandSHA256` in the tests